base/intfuncs.jl

```diff
@@ -460,9 +460,16 @@ function nextpow(a::Real, x::Real)
     a <= 1 && throw(DomainError(a, "`a` must be greater than 1."))
     x <= 1 && return one(a)
     n = ceil(Integer,log(a, x))
+    # round-off error of log can go either direction, so need some checks
     p = a^(n-1)
-    # guard against roundoff error, e.g., with a=5 and x=125
-    p >= x ? p : a^n
+    x > typemax(p) && throw(DomainError(x,"argument is beyond the range of type of the base"))
+    p >= x && return p
+    wp = a^n
+    wp > p || throw(OverflowError("result is beyond the range of type of the base"))
+    wp >= x && return wp
+    wwp = a^(n+1)
```

Comment:
what about `wp*a` powers are relatively expensive.

Reply:
I considered that but it's only valid for integers. For floats there's no guarantee that `x*x^n == x^(n+1)`.

Reply:
yeah, that's unfortunate.

Reply:
For `IEEEFloat` types, bases that are powers of 2 can be done very efficiently with some bit-twiddling. I'll get around to it sooner or later but for now I was focused on correctness.

Reply:
I think you  can possibly do stuff both correct and fast with a modification of the pow implimentation which uses Double-double internally. It would be a bit of a pain though.